core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -654,33 +665,35 @@ trait Schedule_Functions extends Serializable {
    * </pre>
    */
   final def spaced(interval: Duration): Schedule[Any, Int] =
-    forever.delayed(_ + interval)
+    forever.delayed(d => d + relative(interval))
 
   /**
    * A schedule that always recurs, increasing delays by summing the
    * preceding two delays (similar to the fibonacci sequence). Returns the
    * current duration between recurrences.
    */
-  final def fibonacci(one: Duration): Schedule[Any, Duration] =
-    delayed(unfold[(Duration, Duration)]((Duration.Zero, one)) {
-      case (a1, a2) => (a2, a1 + a2)
-    }.map(_._1))
+  final def fibonacci(one: Duration): Schedule[Any, Delay] =
+    delayed(
+      unfold[(Delay, Delay)]((none, relative(one))) {
+        case (a1, a2) => (a2, a1 + a2)
+      }.map(_._1)
+    )
 
   /**
    * A schedule that always recurs, but will repeat on a linear time
    * interval, given by `base * n` where `n` is the number of
    * repetitions so far. Returns the current duration between recurrences.
    */
-  final def linear(base: Duration): Schedule[Any, Duration] =
-    delayed(forever.map(i => base * i.doubleValue()))
+  final def linear(base: Duration): Schedule[Any, Delay] =
+    delayed(forever.map(i => relative(base * i.doubleValue())))
 
   /**
    * A schedule that always recurs, but will wait a certain amount between
    * repetitions, given by `base * factor.pow(n)`, where `n` is the number of
    * repetitions so far. Returns the current duration between recurrences.
    */
-  final def exponential(base: Duration, factor: Double = 2.0): Schedule[Any, Duration] =
-    delayed(forever.map(i => base * math.pow(factor, i.doubleValue)))
+  final def exponential(base: Duration, factor: Double = 2.0): Schedule[Any, Delay] =
```

Comment:
Try to return `Duration` here instead of `Delay`, by running the `Delay` (`Duration` is much easier to work with, and has comparison, equals, hash code, etc.). We might need to add `mapM` to schedule to do that easily.